src/app/src/main/kotlin/org/vaccineimpact/api/app/logic/ExpectationsLogic.kt

```diff
@@ -2,28 +2,26 @@ package org.vaccineimpact.api.app.logic
 
 import org.vaccineimpact.api.app.repositories.ExpectationsRepository
 import org.vaccineimpact.api.app.repositories.ModellingGroupRepository
-import org.vaccineimpact.api.app.repositories.ResponsibilitiesRepository
 import org.vaccineimpact.api.app.repositories.TouchstoneRepository
 import org.vaccineimpact.api.models.Expectations
 
 interface ExpectationsLogic
 {
-    fun getExpectationsForResponsibility(groupId: String,
-                                         touchstoneVersionId: String,
-                                         scenarioId: String): Expectations
+    fun getExpectations(groupId: String,
+                        touchstoneVersionId: String,
+                        expectationId: Int): Expectations
 }
 
-class RepositoriesExpectationsLogic(private val responsibilitiesRepository: ResponsibilitiesRepository,
-                                    private val expectationsRepository: ExpectationsRepository,
+class RepositoriesExpectationsLogic(private val expectationsRepository: ExpectationsRepository,
                                     private val modellingGroupRepository: ModellingGroupRepository,
                                     private val touchstoneRepository: TouchstoneRepository) : ExpectationsLogic
 {
-    override fun getExpectationsForResponsibility(groupId: String, touchstoneVersionId: String, scenarioId: String): Expectations
+    override fun getExpectations(groupId: String, touchstoneVersionId: String, expectationId: Int):
+            Expectations
     {
         modellingGroupRepository.getModellingGroup(groupId) // throws if group does not exist
         touchstoneRepository.touchstoneVersions.get(touchstoneVersionId) // throws if touchstone version does not exist
-        val responsibilityId = responsibilitiesRepository.getResponsibilityId(groupId, touchstoneVersionId, scenarioId)
-        return expectationsRepository.getExpectationsForResponsibility(responsibilityId)
+        return expectationsRepository.getExpectations(expectationId)
```

Comment:
I'm not sure we are currently checking that this expectation belongs to this modelling group and touchstone.

Reply:
Just realised you are doing some refactoring to return expectations with responsibilities, so I'll wait for that to go in as will presumably make this check easier.